tests/README.md

```diff
@@ -182,44 +182,53 @@ expected output. The framework includes the following built-in verify functions:
 1. **`_verify_same`** `func` `expected`
 
    Runs `func` and compares the output with `expected`. If they are not the same,
-   exponentially back off and try again, 7 times by default. The number of retries
-   can be changed by setting the `VERIFY_RETRIES` environment variable.
+   wait a second and try again, up to two minutes by default. The retry behavior
+   can be changed by setting the `VERIFY_TIMEOUT` and `VERIFY_DELAY` environment
```

Comment:
Why do you want to change to a fixed delay, instead of exponential back off? Previously it would only call the function 7 times before giving up in 2 mins, but now it will call the func about 120 times before timing out when there really is a problem.

Reply:
The number of calls isn't really a concern from the perspective of a test author or someone debugging a test ... it's duration (i.e. timeout).  Using timeout also provides more expected behavior in that the test terminates when you expect it to. If using number of attempts, you actually have to factor in how long each attempt takes, which could be several seconds. This means you can have tests that actually take way longer than 2 minutes.

Exponential backoff is more than we need here in general and also leads to confusing behavior WRT the overall duration. In addition, I can't think of anything we would actually want to exponentially backoff ... there's no real advantage. Using a delay and timeout is a much simpler UI from a test authoring perspective.

Reply:
The expected bahavior argument goes both ways. If we don't know how long the function itself will take, we don't know how many retries we'll get before the timeout, so to handle things like curl with network glitches, we can't be sure how many retries we're going to get.

Anyway, I agree that timeout/delay is simpler for people to understand, so OK with the change. I still wonder if ~120 retries by default is too much. Maybe the default delay should be 2 secs?

Reply:
> Maybe the default delay should be 2 secs?

Possibly, but I'm not sure if there is any real concern. If it becomes a problem we can always bump it up.